jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannelOverHttp.java

```diff
@@ -407,6 +407,12 @@ boolean onIdleTimeout(Throwable timeout)
         return true;
     }
 
+    @Override
+    protected void checkAndPrepareUpgrade()
+    {
+        // TODO: move the code from HttpConnection.upgrade() here?
```

Comment:
That would be good to get in this PR... while not strictly related if we don't do it now it will get forgotten